plugins/net_plugin/include/eosio/net_plugin/auto_bp_peering.hpp

```diff
@@ -404,11 +401,11 @@ class bp_connection_manager {
       auto& idx = gossip_bps.index.get<by_producer>();
       bool diff = false;
       for (const auto& peer : msg.peers) {
-         if (auto i = idx.find(boost::make_tuple(peer.producer_name, boost::cref(peer.server_endpoint))); i != idx.end()) {
-            if (i->sig != peer.sig && peer.expiration >= i->expiration) { // signature has changed, producer_name and server_endpoint has not changed
+         if (auto i = idx.find(std::make_tuple(peer.producer_name, std::cref(peer.server_endpoint()))); i != idx.end()) {
```

Comment:
Did we unpack `peer.cached_bp_peer_info` so we can access `peer.server_endpoint()`?

I don't see where, but if we did, why do we inpack it again on line 408?

Reply:
It is unpacked as part of `validate_gossip_bp_peers_message()`.
You are correct, no reason to unpack it again here.